src/test/java/com/microsoft/sqlserver/jdbc/callablestatement/CallableMixedTest.java

```diff
@@ -78,4 +84,44 @@ public void datatypestest() throws Exception {
             }
         }
     }
+
+    @Test
+    @Tag("xAzureSQLDB")
+    @Tag("xAzureSQLDW")
+    @Tag("xAzureSQLMI")
+    public void noPrivilegeTest() throws SQLException {
+        try (Connection c = getConnection();
+                Statement stmt = c.createStatement()) {
+            String tableName = RandomUtil.getIdentifier("jdbc_priv");
+            String procName = RandomUtil.getIdentifier("priv_proc");
+            String user = "priv_user" + UUID.randomUUID();
+            String pass = "priv_pass" + UUID.randomUUID();
+
+
+            stmt.execute(
+                    "CREATE TABLE " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (id int, name varchar(50))");
+            stmt.execute("CREATE PROC " + AbstractSQLGenerator.escapeIdentifier(procName)
+                    + " @id int, @str varchar(50) as INSERT INTO " + AbstractSQLGenerator.escapeIdentifier(tableName)
+                    + " values(@id,@str)");
+            stmt.execute(
+                    "CREATE LOGIN " + AbstractSQLGenerator.escapeIdentifier(user) + " WITH password='" + pass + "'");
+            stmt.execute("CREATE USER " + AbstractSQLGenerator.escapeIdentifier(user) + "");
+            try {
+                stmt.execute("EXECUTE AS USER='" + user + "';EXECUTE " + AbstractSQLGenerator.escapeIdentifier(procName)
+                        + " 1,'hi';");
+                fail(TestResource.getResource("R_shouldThrowException"));
+            } catch (SQLException e) {
+                assertTrue(e.getMessage().matches(TestResource.formatErrorMsg("R_NoPrivilege")));
+            } finally {
+                TestUtils.dropProcedureIfExists(procName, stmt);
+                TestUtils.dropTableIfExists(tableName, stmt);
+                stmt.close();

```

Comment:
No need to explicitly close `stmt` , `c` here, try block will handle that.

Reply:
Need to close to log the connection off so I can drop the credentials in the next block.